src/utils/weather.js

```diff
@@ -1,7 +1,22 @@
 import axios from 'axios';
 
-export const currentWeather = async () => {
-  const url = `http://api.openweathermap.org/data/2.5/weather?lat=30.267153&lon=-97.7430608&appid=${
+const getCurrentPosition = (options = {}) =>
+  new Promise((resolve, reject) => {
+    navigator.geolocation.getCurrentPosition(resolve, reject, options);
+  });
+
+export const getLocation = async () => {
+  let position;
+  try {
+    position = await getCurrentPosition();
+  } catch (error) {
+    console.log(error);
```

Comment:
so if you get an error on getCurrentPosition you will returning undefined, maybe `throw` an exception on the `catch` and use `finally` to return the position